var/spack/repos/builtin/packages/petsc/package.py

```diff
@@ -61,14 +61,24 @@ def mpi_dependent_options(self):
                 errors = ['incompatible variants given'] + errors
                 raise RuntimeError('\n'.join(errors))
         else:
-            compiler_opts = [
-                '--with-mpi=1',
-                '--with-mpi-dir=%s' % self.spec['mpi'].prefix,
-            ]
+            if self.compiler.name == "clang":
+                compiler_opts = [
+                    '--with-mpi=1',
+                    '--with-cc=%s  -Qunused-arguments' % join_path(self.spec['mpi'].prefix.bin, 'mpicc'), # Avoid confusing PETSc config by clang: warning: argument unused during compilation
```

Comment:
because PETSc pass them to fortran compilers as well. See recent discussion on the mailing list. That's not an option, unfortunately. The best solution is to put those in Spack's compiler wrappers, IMHO.